x/stablestake/types/share_denom.go

```diff
@@ -1,5 +1,27 @@
 package types
 
-func GetShareDenom() string {
-	return "stablestake/share"
+import (
+	fmt "fmt"
+	"strconv"
+	"strings"
+)
+
+func GetShareDenomForPool(poolId uint64) string {
+	if poolId == PoolId {
```

Comment:
Since USDC pool has become special pool as its denom share is unique, lets rename PoolId to USDCPoolId ?